### DISCUSSION TITLE
Allow searching relationship fields by entry ID

### DISCUSSION THREAD

**Discussion by swierczek** [ExpressionEngine/ExpressionEngine#3537] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
Searching in a relationship field only works by title, not by entry ID:

<img width="506" alt="image" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/2423727/4bc2bfda-e040-49f6-89f1-e7b33e7ca17e">
<img width="341" alt="image" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/2423727/286faa32-b2ca-4a83-8f03-7d7b8c69813d">

**Describe the solution you'd like**
Be able to search by entry ID or title

**Describe alternatives you've considered**
I was able to add a hook to do the same thing, but IMO this should be core functionality:

```
/**
 * Allow filtering by entry ID in relationship field ajax requests
 *
 * @param QueryBuilder $entries
 * @param int $fieldId
 * @param array $settings
 */
public function relationships_display_field_options(QueryBuilder $entries, int $fieldId, array $settings)
{
    if (isset($settings['search'])) {
        $entries->search('entry_id', $settings['search']);
    }
}
```

**Teachability, Documentation, Adoption, Migration Strategy**
An easy solution would be to adjust the placeholder text to say "Search by title or entry ID" instead of just "Search":

<img width="491" alt="image" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/2423727/86ae0b5a-105f-4c30-b9b0-74cac59cbb4c">

- [x] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by swierczek** [comment]
This is related to https://github.com/ExpressionEngine/ExpressionEngine/pull/1103 which addressed the UI portion, but not the backend functionality of actually searching by entry ID.

I believe it's a quick fix by adjusting the conditional on line 69 of `system/ee/ExpressionEngine/Addons/relationship/libraries/EntryList.php` to be:

Before:
```
if (! empty($search)) {
    $entries->search('title', '"' . $search . '"');
}
```

After:
```
if (! empty($search)) {
    $entries->search('title', $search);
    $entries->search('entry_id', $search);
}
```

**Comment by swierczek** [comment]
I'm not sure where this "Search" placeholder text is defined though.

**Comment by intoeetive** [comment]
Makes perfect sense. In Entry Manager, when we search "titles only" it actually searches title, url_title and entry ID. So we need to do same here fo consistency

https://github.com/ExpressionEngine/ExpressionEngine/pull/3541
